### DISCUSSION TITLE
Where can i find these bubbling fluidized bed reactor cases?

### DISCUSSION THREAD

**Discussion by reactingflow** [IDAES/idaes-pse#1446] [Q&A] [answered]
Hi, everyone, 

I am learning to use IDAES to simulate a bubbling fluidized bed reactor. When i search for the keywords 'fluidized bed' on the IDAES website, it shows that there are two cases in IDAES, however, i can't find them. Could you please tell me whether they are accessible and where i can find them? 

Thank you!

![1](https://github.com/IDAES/idaes-pse/assets/151173251/23c8643f-4e39-4f5b-b266-64bbf64b75c1)

**Comment by andrewlee94** [comment, marked as answer]
All of the code for these should be here: https://github.com/IDAES/idaes-pse/tree/main/idaes/models_extra/gas_solid_contactors

**Reply by reactingflow** [reply]
Hi, Dr. Lee,

I recently read your paper published in **Industrial & Engineering Chemistry Research** related to this fluidized bed model. I'd like to know if the heat transfer process between the phases and the heat exchange tubes, as mentioned in the paper, can also be implemented using this IDAES model, since I couldn't find any related descriptions in the documentation.

 Thank you!

**Reply by andrewlee94** [reply]
It is definitely possible to implement the equations for the in-bed heat exchanger, however that is a significant amount of effort and we do not have the time or scope to implement that ourselves at this time. If you happen to have the time yourself, we would welcome contributions to add those equations but it would require a lot of work to implement and test the code.

**Reply by reactingflow** [reply]
Hi, Dr. Lee,

Thank you for your response. We are willing to contribute our code if it can be accepted. Recently, we have been attempting to model heat exchangers within fluidized beds. For large fluidized beds, there can be many different arrangements for the heat exchange tubes within the bed. We have reviewed papers and handbooks but have not yet found a satisfactory modeling approach. Could you provide us with some insights? Thank you!

Additionally, we noticed that the fluidized bed model in IDAES differs significantly from the Kunii-Levenspiel model, although the basic ideas are similar. In the online reference guides, we found formulas that were not mentioned in the papers, and we do not understand why there is a factor of 6 in these formulas. Moreover, the dimensions of the two terms in the formula  seem inconsistent. 

![1](https://github.com/user-attachments/assets/8e03c6a1-e0bf-4938-867a-3b03672d6e62)

**Reply by andrewlee94** [reply]
@reactingflow Unfortunately it has been many years since I worked on that model so my memories of the details are a bit vague. The best reference I can give is this one, which was the original paper we wrote on the development of the model:

A. Lee, D.C. Miller. A one-dimensional (1-D) three-region model for a bubbling
fluidized-bed Adsorber, Ind. Eng. Chem. Res. 52 (2013) 469–484.

There is also a book by Kunii and Levenspiel on Fluidization that served as our main reference when developing the model (notably this is a bit different from the original paper from memory).

Regarding the differences from the Kunii-Levenspiel model, we took correlations from a number of other references to better model some of the phenomena in the bed. The core structure of the model is that of Kunii and Levenspiel, but many of the correlations for bed characteristics (velocities and voidages) come from experimental papers on the subject.
